### PR TITLE
[Merged by Bors] - feat(Order/Cover): `covBy_iff_lt_iff_le`

### PR DESCRIPTION
**Pull request by vihdzp**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Cover.lean [hunk 1]
```diff
@@ -421,6 +421,19 @@ lemma LT.lt.exists_disjoint_Iio_Ioi (h : a < b) :
...
+  mpr := fun H ↦ ⟨(H _).2 le_rfl, fun _z hx hz ↦ ((H _).1 hz).not_lt hx⟩
+
+theorem covBy_iff_le_iff_lt {x y : α} : x ⋖ y ↔ ∀ z, z ≤ x ↔ z < y := by
+  simp_rw [covBy_iff_lt_iff_le, iff_comm]
+
+theorem CovBy.lt_iff_le {x y z : α} (h : x ⋖ y) : z < y ↔ z ≤ x :=
+  covBy_iff_lt_iff_le.1 h z
+
+theorem CovBy.le_iff_lt {x y z : α} (h : x ⋖ y) : z ≤ x ↔ z < y :=
+  covBy_iff_le_iff_lt.1 h z
```
Comment (YaelDillies) [inline Mathlib/Order/Cover.lean:435]: Can you declare these using `alias`?

File: Mathlib/Order/Cover.lean [hunk 2]
```diff
@@ -421,6 +421,19 @@ lemma LT.lt.exists_disjoint_Iio_Ioi (h : a < b) :
   · rcases h.exists_lt_lt h' with ⟨c, ha, hb⟩
     exact ⟨c, ha, c, hb, fun _ h₁ _ => lt_trans h₁⟩
 
+theorem covBy_iff_lt_iff_le {x y : α} : x ⋖ y ↔ ∀ z, z < y ↔ z ≤ x where
+  mp := fun hx _z ↦ ⟨hx.le_of_lt, fun hz ↦ hz.trans_lt hx.lt⟩
+  mpr := fun H ↦ ⟨(H _).2 le_rfl, fun _z hx hz ↦ ((H _).1 hz).not_lt hx⟩
```
Comment (YaelDillies) [inline Mathlib/Order/Cover.lean:426]: Can we have `x ⋖ y ↔ ∀ z, x < z ↔ y ≤ z`?

File: Mathlib/Order/Cover.lean [hunk 3]
```diff
@@ -421,6 +421,19 @@ lemma LT.lt.exists_disjoint_Iio_Ioi (h : a < b) :
...
+
+theorem covBy_iff_le_iff_lt {x y : α} : x ⋖ y ↔ ∀ z, z ≤ x ↔ z < y := by
+  simp_rw [covBy_iff_lt_iff_le, iff_comm]
```
Comment (YaelDillies) [inline Mathlib/Order/Cover.lean:429]: Not sure we need both this one and the one above

Comment (vihdzp) [reply Mathlib/Order/Cover.lean:429]: The way I want to use this lemma is to rewrite the expression on the right. But I think it's perfectly reasonable to expect it to be the other way around.

File: Mathlib/Order/Cover.lean (hunk 1 above)
Comment (vihdzp) [reply Mathlib/Order/Cover.lean:435]: The argument implicitness of z varies, but conversely, I don't know if it should also be implicit in the iff lemmas.

Comment (YaelDillies) [reply Mathlib/Order/Cover.lean:435]: I would think it should

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**